packages/x-charts/src/ChartsOverlay/ChartsLoadingOverlay.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import { useDrawingArea } from '../hooks/useDrawingArea';
+import type { CommonOverlayProps } from './ChartsOverlay';
+
+const StyledText = styled('text')(({ theme }) => ({
+  stroke: 'none',
+  fill: theme.palette.text.primary,
+  shapeRendering: 'crispEdges',
+  textAnchor: 'middle',
+  dominantBaseline: 'middle',
+}));
+
+export function ChartsLoadingOverlay(props: CommonOverlayProps) {
+  const { top, left, height, width } = useDrawingArea();
+
+  return (
+    <StyledText x={left + width / 2} y={top + height / 2} {...props}>
+      Loading data ...
```

Comment:
I guess you don't want to introduce a localization system for this text :laughing: 
But people should be able to pass `slotProps={{ loadingOverlay: { children: 'Chargement des données' } }}` and right now I don't think it works 

The following might do the trick:

```suggestion
      props.children ?? 'Loading data ...'
```

Reply:
Sorry I didn't see it was in draft :angel:

Reply:
No issue I was motivated to introduce a localization system, but the notion of props might be better as long as we do not see other translation needs

Reply:
> as long as we do not see other translation needs

I agree with your approach
If this will likely be the only text in the foreseeable future, a prop is fine
If we think of some other texts that might be introduced soon, the localization system is required